src/modules/new-request-form/usePrefilledTicketFields.tsx

```diff
@@ -0,0 +1,88 @@
+import type { Field } from "./data-types";
+import DOMPurify from "dompurify";
+
+const MAX_URL_LENGTH = 2048;
+const TICKET_FIELD_PREFIX = "tf_";
+const ALLOWED_SYSTEM_FIELDS = [
+  "anonymous_requester_email",
+  "priority",
+  "type",
+  "description",
+  "subject",
+  "due_at",
+  "collaborators",
+  "organization_id",
+];
+const ALLOWED_BOOLEAN_VALUES = ["true", "false"];
+const ALLOWED_HTML_TAGS = [
+  "pre",
+  "strong",
+  "b",
+  "p",
+  "blockquote",
+  "ul",
+  "ol",
+  "li",
+  "h2",
+  "h3",
+  "h4",
+  "i",
+  "em",
+  "br",
+];
+
+export function usePrefilledTicketFields(fields: Field[]): Field[] {
+  const { href } = location;
+  const params = new window.URL(href).searchParams;
+
+  if (href.length > MAX_URL_LENGTH) return fields;
+  if (params.get("parent_id")) return fields;
+
+  for (const [key, value] of params) {
+    if (!key.startsWith(TICKET_FIELD_PREFIX)) continue;
+
+    const ticketFieldId = key.substring(TICKET_FIELD_PREFIX.length);
+    const isSystemField = ALLOWED_SYSTEM_FIELDS.includes(ticketFieldId);
+    const isCustomField = Number(ticketFieldId);
```

Comment:
Nit: This assignment is not quite easy to read, the `is` prefix suggests it is a boolean, and it is used as a boolean in the following statement, but the value is actually a number or NaN.

I know it is ported from the current implementation, and I am okay with keeping it like this, but maybe something like `const isCustomField = !Number.isNaN(Number(ticketFieldId));` would be clearer.